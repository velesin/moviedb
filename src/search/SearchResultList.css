.SearchResultList {
    padding-bottom: 64px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
}