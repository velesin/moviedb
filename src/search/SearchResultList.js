import React from 'react';
import { SearchResultListItem } from './SearchResultListItem';

export function SearchResultList({ movies }) {
    return (
        <div className="SearchResultList">
            { movies.map(movie => (
                <SearchResultListItem
                    key={ movie.id }
                    movie={ movie }
                />
            )) }
        </div>
    );
}