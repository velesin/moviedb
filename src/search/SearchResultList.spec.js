import React from 'react';
import { shallow } from 'enzyme';

import { SearchResultList } from './SearchResultList';

const MOVIES = [ { id: 1 }, { id: 2 }, { id: 3 } ];
let list;

beforeEach(() => {
    list = shallow(<SearchResultList movies={ MOVIES } />);
});

it('renders correctly a list with movies', () => {
    expect(list.getElement()).toMatchSnapshot();
});

it('assigns correct keys to items in the list', () => {
    const keys = list.find('SearchResultListItem').map(item => item.key());
    expect(keys).toEqual([ '1', '2','3' ]);
});