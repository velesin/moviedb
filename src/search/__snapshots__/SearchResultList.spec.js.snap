// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a list with movies 1`] = `
<div
  className="SearchResultList"
>
  <SearchResultListItem
    movie={
      Object {
        "id": 1,
      }
    }
  />
  <SearchResultListItem
    movie={
      Object {
        "id": 2,
      }
    }
  />
  <SearchResultListItem
    movie={
      Object {
        "id": 3,
      }
    }
  />
</div>
`;
